packages/docregistry/test/registry.spec.ts

```diff
@@ -700,6 +700,42 @@ describe('docregistry/registry', () => {
         expect(ft.name).toBe('notebook');
       });
 
+      it('allow customise filetype for directory', () => {
+        registry.addFileType({
+          name: 'node_module',
+          contentType: 'directory',
+          pattern: '^node_modules$'
+        });
+        const regularDirectoryFt = registry.getFileTypeForModel({
+          path: '/foo',
+          type: 'directory'
+        });
+        expect(regularDirectoryFt.name).toBe('directory');
+        const nodeNoduleFt = registry.getFileTypeForModel({
+          path: '/foo/node_modules',
+          type: 'directory'
+        });
+        expect(nodeNoduleFt.name).toBe('node_module');
```

Comment:
Just for clarity for a future reader:

```suggestion
        const nodeModuleFt = registry.getFileTypeForModel({
          path: '/foo/node_modules',
          type: 'directory'
        });
        expect(nodeModuleFt.name).toBe('node_module');
```